numpy/_core/src/multiarray/nditer_api.c

```diff
@@ -1924,17 +1954,25 @@ npyiter_copy_from_buffers(NpyIter *iter)
         return 0;
     }
 
-    NPY_IT_DBG_PRINT("Iterator: Copying buffers to outputs\n");
-
-    if (itflags&NPY_ITFLAG_REDUCE) {
-        reduce_outerdim = NBF_REDUCE_OUTERDIM(bufferdata);
-        reduce_outerstrides = NBF_REDUCE_OUTERSTRIDES(bufferdata);
-        reduce_outeraxisdata = NIT_INDEX_AXISDATA(axisdata, reduce_outerdim);
+    if (itflags & NPY_ITFLAG_REDUCE) {
+        npy_intp sizeof_axisdata = NIT_AXISDATA_SIZEOF(itflags, ndim, nop);
+        outer_axisdata = NIT_INDEX_AXISDATA(axisdata, bufferdata->outerdim);
```

Comment:
Not your problem, but, really, having macros require other variables to be defined is pretty bad form! Especially since here `axisdata_incr` could have made use of `sizeof_axisdata`.